### ISSUE TITLE
[Enhancement]: Cache `FromDockerfile` builds to allow multiple containers using the same image

### ISSUE THREAD

**Issue by turt2live** [testcontainers/testcontainers-go#1485]
### Proposal

When using `FromDockerfile`, the image is built *every* time the container is requested, but the contents of that image are unlikely to change. Being able to specify a cache key or something to avoid a secondary build would be appreciated. 

See https://github.com/testcontainers/testcontainers-go/issues/1484 for how I've worked around this already. The following being the code I've used:

```go
var mmrCachedImage string
var mmrCachedContext *os.File

func reuseMmrBuild(ctx context.Context) (string, error) {
	if mmrCachedImage != "" {
		return mmrCachedImage, nil
	}
	log.Println("[Test Deps] Building MMR image...")
	cr, err := createDockerContext()
	if err != nil {
		return "", err
	}
	mmrCachedContext = cr
	buildReq := testcontainers.GenericContainerRequest{
		ContainerRequest: testcontainers.ContainerRequest{
			FromDockerfile: testcontainers.FromDockerfile{
				Dockerfile:     "Dockerfile",
				Context:        ".",
				ContextArchive: cr,
				PrintBuildLog:  true,
			},
		},
		Started: false,
	}
	provider, err := buildReq.ProviderType.GetProvider(testcontainers.WithLogger(testcontainers.Logger))
	if err != nil {
		return "", err
	}
	c, err := provider.CreateContainer(ctx, buildReq.ContainerRequest)
	if err != nil {
		return "", err
	}
	if dockerC, ok := c.(*testcontainers.DockerContainer); !ok {
		return "", errors.New("failed to convert built MMR container to a DockerContainer")
	} else {
		mmrCachedImage = dockerC.Image
	}
	log.Println("[Test Deps] Cached build as ", mmrCachedImage)
	return mmrCachedImage, nil
}
```

Note how it effectively gutwrenches the image name out of the container so it can reuse it later as `testcontainers.ContainerRequest{Image: mmrImage}`.

**Comment by mdelapenya** [comment]
@turt2live could you try with the latest release? #1785 was merged in v0.26.0, so it should be possible now.

Thanks!

**Comment by turt2live** [comment]
oooh, absolutely will give it a go. Thank you!
